tools/offcputime

```diff
@@ -82,32 +88,32 @@ static u64 get_frame(u64 *bp) {
     return 0;
 }
 
-int offcpu(struct pt_regs *ctx) {
-    u32 pid = bpf_get_current_pid_tgid();
-    u64 ts = bpf_ktime_get_ns();
-    FILTER
-    start.update(&pid, &ts);
-    return 0;
-}
+int oncpu(struct pt_regs *ctx, struct task_struct *prev) {
+    u32 pid;
+    u64 ts, *tsp;
 
-int oncpu(struct pt_regs *ctx) {
-    u32 pid = bpf_get_current_pid_tgid();
-    FILTER
-    u64 ts = bpf_ktime_get_ns();
-    struct key_t key = {};
-    u64 zero = 0, *val, bp = 0, *tsp, delta;
-    int depth = 0;
+    // record previous thread sleep time
+    if (FILTER) {
+        pid = prev->pid;
```

Comment:
nice. lower overhead.
do you think it makes sense to add key.pid,
since key.comm muxes all threads of given process into one
and for multi threaded apps the final numbers are hard to interpret.
Also '-p pid' is also useless for such apps, since it filters by kernel pid,
but user spaces guys think of tgid when they say pid